lib/TargetBuilder.js

```diff
@@ -0,0 +1,58 @@
+var regex = new RegExp(/(:[^/-]+)/)
+  , path = require('path')
+
+var reservedKeysForModel = {
```

Comment:
4 - 28, seems complicated, but it makes 40-43 simple. Any refactoring suggestions?